code.js

```diff
@@ -1,3 +1,36 @@
-function quicksort(array) {
+function quicksort(array = []) {
+    if (array.length <= 1){
+        return array;
+    }
+    numSort = Array(array.length).fill(0); // This array tells our algorithm what numbers are alreay storted
+    // movePivot(numSort, array, 0, 1);
+    for (pivot = 0; numSort.indexOf(0) != -1; pivot = 0){
+        placed = false;
+        for(pivot; placed == false; pivot ++){
+            if (numSort[pivot] == 0){
+                lessThan = pivot + 1; // Stores the index of the last item less than our pivot
+                pivot = movePivot(numSort, array, pivot, lessThan);
+                placed = true;
+                console.log(numSort);
+            }
+        }
+    } 
     return array;
 }
+
+function movePivot(numSort, array, pivot, lessThan) {
+    for(j = pivot+1; numSort[j] == 0 && j < array.length; j++){
```

Comment:
This looks like you're only processing elements to the right of the pivot. Where are you doing the part left of the pivot?

Reply:
The way that my for loops control `movePivot` ensures that I'm processing to the right and left of the pivot I just placed.

In the first iteration it places the first pivot hopefully somewhere in the middle of the list and returns the value after the end of the elements we just processed to the loop, this ensures it only processes elements to the left and right of the pivot once, and here it stops and a new iteration starts. In the new iteration it sorts the pivot to the right of the first pivot, and then `movePivot` returns the number after the end of the elements we processed , which would be 1, the number we sorted previously. And then my algorithm sorts the left side because `movePivot` kicked pivot in my inside for loop to the the other side of the previous pivot, which is now doing it recursively haha ':D